microscope/batch/batch.py

```diff
@@ -1,17 +1,31 @@
 import time
-import queue as queuemodule
 import sys
 
+import queue as queuemodule
+
 from microscope.monitor.monitor import MonitorRunner
 
 
 def batch(runner: MonitorRunner, timeout: int):
     start_time = time.time()
     while(runner.is_alive() and runner.close_queue.empty()
           and (start_time + timeout > time.time() or timeout == 0)):
+        while True:
+            try:
+                output = runner.data_queue.get(True, 1)
+                if ("output" in output):
+                    print(f"\n{output['node_name']}: {output['output']}",
+                          end="")
+            except queuemodule.Empty:
+                break
+
+    # drain queue
```

Comment:
Does this loop ever drain data? if we're breaking on `queuemodule.Empty` it would remain empty, wouldn't it? if it was a racey thing, wouldn't it be better to not break on the first empty but sleep/try again in the loop above? This way we do the printing in one spot. I may have grossly misunderstood what this function is doing, though!

Reply:
The while above runs until `runner.close_queue` is empty, output messages are sent via `runner.data_queue`.

Reply:
Hmm, could you explain how it works? because we did the `.get` on `runner.data_queue` I figured the exception it handled was when `runner.data_queue` was empty. The loop below (line 23) also does the get on `data_queue`. I guess somewhere else there's some linkage between the two queues?